cmd/kube-scheduler/app/server.go

```diff
@@ -160,20 +160,6 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 		checks = append(checks, cc.LeaderElection.WatchDog)
 	}
 
-	// Start up the healthz server.
-	if cc.InsecureServing != nil {
-		separateMetrics := cc.InsecureMetricsServing != nil
-		handler := buildHandlerChain(newHealthzHandler(&cc.ComponentConfig, separateMetrics, checks...), nil, nil)
-		if err := cc.InsecureServing.Serve(handler, 0, ctx.Done()); err != nil {
-			return fmt.Errorf("failed to start healthz server: %v", err)
-		}
-	}
-	if cc.InsecureMetricsServing != nil {
```

Comment:
@liggitt @deads2k @soltysh is this saying that scheduler metrics are only being served via insecure ? the `newMetricsHandler` is unused after this change... maybe I read this wrong.

Reply:
Spoke with @mfojtik on slack, it's properly handled few lines below in `newHealthzHandler` the only difference is for insecure it's possible to specify separate addresses for healthz and metrics, for secure both are served from a single address.